src/commands/buffer.ts

```diff
@@ -37,19 +37,42 @@ export async function execute(
     let format: string[] = []
 
     for (const message of messages) {
+      limitMessageLength(message, 4000)
       format.push(`${formatMessage(message)}\n`)
-      if (format.join("").length > 4096) {
-        format.shift()
+      while (format.join("").length > 4000) {
+        removeLongestString(format)
       }
     }
-
     embed.setDescription(format.join(""))
   } else {
     embed.setDescription("No messages found.")
   }
   msg.reply({ embeds: [embed] })
 }
 
+function limitMessageLength(message: ChatBufferMessage, maxLength: number) {
+  if (message.content.length > maxLength) {
+    message.content = message.content.substring(0, maxLength)
+  }
+  return
```

Comment:
make sure you return the substring instead of modifying the message.content

some functions might rely on an unchanged message.content (and other functions might be affected by this too in the future)